centreon/src/Core/HostTemplate/Application/UseCase/PartialUpdateHostTemplate/PartialUpdateHostTemplate.php

```diff
@@ -185,103 +187,42 @@ private function updateMacros(PartialUpdateHostTemplateRequest $request, HostTem
          */
         [$directMacros, $inheritedMacros, $commandMacros] = $this->findOriginalMacros($hostTemplate);
 
-        $macros = $this->createMacros($request->macros, $hostTemplate->getId(), $directMacros, $inheritedMacros);
+        $macros = [];
+        foreach ($request->macros as $data) {
+            $macro = HostMacroFactory::create($data, $hostTemplate->getId(), $directMacros, $inheritedMacros);
+            $macros[$macro->getName()] = $macro;
+        }
 
         $hostMacroDifference = new HostMacroDifference();
-        $macrosDifference = $hostMacroDifference->compute($directMacros, $inheritedMacros, $commandMacros, $macros);
+        $macrosDiff = $hostMacroDifference->compute($directMacros, $inheritedMacros, $commandMacros, $macros);
 
-        foreach ($macrosDifference['removed'] as $macro) {
+        $macrosDiff = MacroManagement::setOrder($macros, $macrosDiff, $directMacros);
```

Comment:
Design consistency problems between `HostMacroDifference` which has non-static methods and `MacroManagement` which has static methods.